File: core/src/com.tuxbear.dinos/domain/game/Wall.java
package com.tuxbear.dinos.domain.game;

/**
 * Created by Ole - André Johansen on 02.01.14.
 */
public class Wall {

    private BoardPosition from;
    private BoardPosition to;

    public Wall() {
    }

    public Wall(int fromX, int fromY, int toX, int toY){
        from = new BoardPosition(fromX, fromY);
        to = new BoardPosition(toX, toY);
    }

    public BoardPosition getTo() {
        return to;
    }

    public void setTo(BoardPosition to) {
        this.to = to;
    }

    public BoardPosition getFrom() {
        return from;
    }

    public void setFrom(BoardPosition from) {
        this.from = from;
    }
}

File: domain/src/com/tuxbear/dinos/domain/game/Wall.kt
package com.tuxbear.dinos.domain.game

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBDocument

@DynamoDBDocument
class Wall {
    @DynamoDBAttribute
    var from: BoardPosition? = null

    @DynamoDBAttribute
    var to: BoardPosition? = null

    constructor(fromX:Int, fromY:Int, toX:Int, toY:Int)
            : this(BoardPosition(fromX, fromY), BoardPosition(toX, toY))

    constructor(from: BoardPosition, to: BoardPosition) {
        this.from = from
        this.to = to
    }

    constructor()
}